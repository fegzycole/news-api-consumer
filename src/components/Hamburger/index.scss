@import "../../styles/mixins";

@mixin get-top-and-width($top, $width) {
  top: $top;
  width: $width;
}

@mixin hamburger-basics {
  position: absolute;
  height: 3px;
  background: $primary-color;
}

@mixin hamburger-basics-content {
  @include hamburger-basics;
  content: "";
}

.Hamburger {
  left: 1rem;
  @include get-top-and-width(2rem, 16px);
  @include hamburger-basics;

  &::before {
    @include get-top-and-width(-8px, 8px);
    @include hamburger-basics-content;
  }

  &::after {
    @include get-top-and-width(8px, 12px);
    @include hamburger-basics-content;
  }

  @include lgScreen {
    display: none;
  }
}
