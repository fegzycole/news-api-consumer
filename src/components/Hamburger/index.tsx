import './index.scss';

const Hamburger = () => {
  return <span className="Hamburger" />;
};

export default Hamburger;
