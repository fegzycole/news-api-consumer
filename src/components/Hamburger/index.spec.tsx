import { render } from "@testing-library/react";

import Hamburger from "./index";

describe("<Hamburger />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match the Hamburger component snapshot", () => {
    const view = render(<Hamburger />);
    expect(view.container).toMatchSnapshot();
  });
});
