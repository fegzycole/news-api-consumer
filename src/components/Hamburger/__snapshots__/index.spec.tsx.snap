// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> should match the Hamburger component snapshot 1`] = `
<div>
  <span
    class="Hamburger"
  />
</div>
`;
